### PR TITLE
Issue 1104: Log GC events from segment store and controller.

### LINKED ISSUE #1104: Log GC events from segment store and controller processes
**Problem description**
GC event could delay responses from segment store and controller. In recent system tests we observed significant delay between event send and event receive. One of the reasons for this could be GC happening in controller or relevant segment store process. Logging GC events would help correlate delay in responses with GC events in controller and segment store processes.

**Problem location**
Startup command of segment store and controller.

**Suggestions for an improvement**
Add GC event logging while starting segment store and controller processes, e.g., `-XX:+PrintGCDetails -XX:+PrintGCDateStamps -Xloggc:/opt/app/gc.log`

### PR DESCRIPTION
**Pull request by shrids**
**Change log description**
* Enable GC event logging.
* GC logs are piped to file gc.log under logs/ directory.
* Enable rotation of GC logs. 

**Purpose of the change**
Fixes #1104 

**What the code does**
Enables GC event logging.

**How to verify it**
Ran controller locally to verify gc.log file creation.

### CODE REVIEW

**Comment by skrishnappa** [comment]
@shrids can you rebase your fork/branch to the latest pravega master. I guess the code coverage reduction is due to this.

File: build.gradle [hunk 1]
```diff
@@ -337,6 +340,9 @@ project('controller') {
     applicationName = "pravega-controller"
     mainClassName = "io.pravega.controller.server.Main"
     applicationDefaultJvmArgs = ["-server", "-Xms128m", "-XX:+HeapDumpOnOutOfMemoryError",
+                                 "-XX:+PrintGCDetails", "-XX:+PrintGCDateStamps",
+                                 "-Xloggc:PRAVEGA_APP_HOME/logs/gc.log", "-XX:+UseGCLogFileRotation",
```
Comment (skrishnappa) [inline build.gradle]: This path seems same for both controller and segmentstore, I guess not an issue with multi-node or when deployed on containers. Please review this for singlenode and rename if necessary.

Comment (shrids) [inline build.gradle]: We have have a single docker image for controller and segmentstore. The path of the logs are the same for both of them, inside the container i.e. PRAVEGA_APP_HOME/logs which maps to `/opt/pravega/`.
Also, gc logs are not enabled for pravega singlenode ( The start up script for single node is different from that of controller and SS)

**Comment by fpj** [comment]
This is useful, but I'm worried about making GC logging the default. The default should really be a minimum amount of logging. Could we have it in our gradle configuration, but not enabled by default?

**Comment by shrids** [comment]
Updated the PR to ensure the following
a. GC logs are not enabled by default. i.e. startup scripts `/bin/pravega-controller `and `/bin/pravega-segmentstore` do not generate GC logs
b. GC logs are enabled as part of separate scripts. i.e. to run controller and SSS with GC logs we can use the following startup scripts `bin/pravega-controllerWithGCLog `and `bin/pravega-segmentstoreWithGCLog`

**Pull request merged**